src/NzbDrone.Core/Indexers/Definitions/Newznab/NewznabRssParser.cs

```diff
@@ -38,7 +38,7 @@ public static void CheckError(XDocument xdoc, IndexerResponse indexerResponse)
             var code = Convert.ToInt32(error.Attribute("code").Value);
             var errorMessage = error.Attribute("description").Value;
 
-            if (code >= 100 && code <= 199)
+            if (code is >= 100 and <= 199 or 403)
```

Comment:
https://drunkenslug.com/api?t=search&apikey=1233
`<error code="403" description="Incorrect user credentials (wrong API key)"/>`

https://api.nzbgeek.info/api?t=search&apikey=1233
<error code="100" description="Invalid API Key"/>

may be specific to slug, but if others use it - doesn't hurt.

Reply:
403, so this should get handled fine anyway?
![image](https://github.com/Prowlarr/Prowlarr/assets/376117/b1898887-d2cc-4273-ade7-94de6ccbfc31)

Reply:
403 is not handled anywhere and would throw as unexpected.

I'll reach out to them since they should be returning an HTTP 200 and code 100 per spec.

Reply:
Current and PR State do not have any special handling for HTTP 403

https://gist.github.com/bakerboy448/7e455a42d7796ba8059118a0c3a425b4

```
2023-05-17 21:23:36.3|Warn|Newznab|HTTP Error - Res: [GET] https://drunkenslug.com/api?t=search&extended=1&apikey=(removed) 403.Forbidden (116 bytes)
<?xml version="1.0" encoding="UTF-8"?>
<error code="403" description="Incorrect user credentials (wrong API key)"/>

2023-05-17 21:23:36.3|Trace|Newznab|Unexpected Response content (116 bytes): <?xml version="1.0" encoding="UTF-8"?>
<error code="403" description="Incorrect user credentials (wrong API key)"/>

2023-05-17 21:23:36.3|Warn|Newznab|Unable to connect to indexer
```
which ends up returning the generic unexpected status code
Unable to connect to indexer. Indexer API call resulted in an unexpected StatusCode [Forbidden]

Reply:
But error seems pretty clear in that production example, what exactly are we trying to solve? `Forbidden` (403) is the status.

You are saying `Unable to connect to indexer. Indexer API call resulted in an unexpected StatusCode [Forbidden]` isn't clear enough?

Reply:
> Then treat the status code, not code from body.

The body is what has the details of the response to pass back to the user...

> You are saying `Unable to connect to indexer. Indexer API call resulted in an unexpected StatusCode [Forbidden]` isn't clear enough?

Correct.  To the users that means nothing.  

Though if slug's http response was per spec this would be an non-issue.
If it was a HTTP 200 then we wouldn't need the conditional non-html content passthrough for 401/403.
If it was a HTTP 200 AND had the spec error of `100` we wouldn't need to handle the custom 403 error code.